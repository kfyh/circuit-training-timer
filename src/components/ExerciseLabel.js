import React from 'react'
import PropTypes from 'prop-types'

function ExerciseLabel (props) {
  const interval = props.interval
  const excerciseLabel = interval.label
  const totalReps = interval.reps

  return (
    <div>{excerciseLabel} ({props.currentRep}/{totalReps})</div>
  )
}

ExerciseLabel.propTypes = {
  interval: PropTypes.object,
  currentRep: PropTypes.number
}

export default ExerciseLabel
