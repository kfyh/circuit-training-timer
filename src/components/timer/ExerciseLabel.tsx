import React, { ReactElement } from 'react';
import { FlattenedStep } from '../../types/circuits';

type ExerciseLabelProps = {
	step: FlattenedStep;
};

export function ExerciseLabel(props: ExerciseLabelProps): ReactElement {
	const step = props.step;
	const excerciseLabel = step.label;
	const totalReps = step.repTotal;

	return (
		<div>
			{step.parentLabel && (
				<p>
					{step.parentLabel} ({step.parentRepIndex}/{step.parentRepTotal})
				</p>
			)}
			<p>
				{excerciseLabel} ({props.step.repIndex}/{totalReps})
			</p>
		</div>
	);
}
