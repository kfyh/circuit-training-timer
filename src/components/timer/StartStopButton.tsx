import React, { ReactElement } from 'react';

type StartStopButtonProps = {
	isRunning: boolean;
	onClick: () => void;
}

export function StartStopButton({ isRunning, onClick }: StartStopButtonProps): ReactElement {
	const label = isRunning ? 'Stop' : 'Start';
	return <button onClick={onClick}>{label}</button>;
}
