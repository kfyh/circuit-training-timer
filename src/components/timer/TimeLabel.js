import React from 'react';
import PropTypes from 'prop-types';

export function TimeLabel({ time }) {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	let displayTime = Math.ceil(Math.max(0, time / 1000));
	if (displayTime > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}

TimeLabel.propTypes = {
	time: PropTypes.number,
};
