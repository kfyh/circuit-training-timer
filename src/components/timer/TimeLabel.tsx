import React, { ReactElement } from 'react';

type TimeLabelProps = {
	time: number;
	isResting: boolean;
};

export function TimeLabel({ time, isResting }: TimeLabelProps): ReactElement {
	const divStyle = {
		color: 'blue',
		fontSize: '250pt',
	};

	if (isResting) {
		divStyle.color = 'green';
	}

	const timeSeconds = Math.ceil(Math.max(0, time / 1000));
	let displayTime = timeSeconds.toString();
	if (timeSeconds > 60) {
		displayTime = Math.ceil(time / 1000 / 60) + ' mins';
		divStyle.fontSize = '130pt';
	}

	return <div style={divStyle}>{displayTime}</div>;
}
