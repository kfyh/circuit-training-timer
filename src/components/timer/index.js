export * from './StartStopButton';
export * from './ExerciseLabel';
export * from './TimeLabel';
