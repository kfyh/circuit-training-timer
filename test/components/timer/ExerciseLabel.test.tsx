import React from 'react';
import { shallow } from 'enzyme';
import { ExerciseLabel } from '../../../src/components/timer/ExerciseLabel';

describe('Given ExerciseLabel', () => {
	const testValues = [
		[{ id: '0', label: 'Burpies', time: 60, repIndex: 1, repTotal: 3, type: 'exercise' }, 1, 'Burpies (1/3)'],
		[{ id: expect.anything(), label: 'Push Ups', time: 60, repIndex: 2, repTotal: 5, type: 'exercise' }, 1, 'Push Ups (2/5)'],
		[
			{
				id: expect.anything(),
				label: 'Hang for 7 secs',
				time: 7,
				repIndex: 1,
				repTotal: 1,
				type: 'exercise',
				parentId: expect.anything(),
				parentLabel: 'Finger Training',
				parentType: 'set',
				parentRepIndex: 2,
				parentRepTotal: 5,
			},
			1,
			'',
		],
	];

	test.each(testValues)('When interval is %o, rep is %i then label is %s', (step, currentRep, expected) => {
		const render = shallow(<ExerciseLabel step={step} />);
		expect(render).toMatchSnapshot();
	});
});
