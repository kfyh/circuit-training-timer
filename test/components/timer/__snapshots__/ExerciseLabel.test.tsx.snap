// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given ExerciseLabel When interval is { id:
   Anything {
     sample: undefined,
     '$$typeof': Symbol(jest.asymmetricMatcher),
     inverse: undefined },
  label: 'Hang for 7 secs',
  time: 7,
  repIndex: 1,
  repTotal: 1,
  type: 'exercise',
  parentId:
   Anything {
     sample: undefined,
     '$$typeof': Symbol(jest.asymmetricMatcher),
     inverse: undefined },
  parentLabel: 'Finger Training',
  parentType: 'set',
  parentRepIndex: 2,
  parentRepTotal: 5 }, rep is 1 then label is  1`] = `
<div>
  <p>
    Finger Training
     (
    2
    /
    5
    )
  </p>
  <p>
    Hang for 7 secs
     (
    1
    /
    1
    )
  </p>
</div>
`;

exports[`Given ExerciseLabel When interval is { id:
   Anything {
     sample: undefined,
     '$$typeof': Symbol(jest.asymmetricMatcher),
     inverse: undefined },
  label: 'Push Ups',
  time: 60,
  repIndex: 2,
  repTotal: 5,
  type: 'exercise' }, rep is 1 then label is Push Ups (2/5) 1`] = `
<div>
  <p>
    Push Ups
     (
    2
    /
    5
    )
  </p>
</div>
`;

exports[`Given ExerciseLabel When interval is { id: '0',
  label: 'Burpies',
  time: 60,
  repIndex: 1,
  repTotal: 3,
  type: 'exercise' }, rep is 1 then label is Burpies (1/3) 1`] = `
<div>
  <p>
    Burpies
     (
    1
    /
    3
    )
  </p>
</div>
`;
